File: tinysoul/endpoint/http/schemas/maintenance.py
"""Maintenance request schemas."""

from __future__ import annotations

from typing import Literal

from pydantic import BaseModel, ConfigDict, Field

from tinysoul.infra.json import JsonValue


class MaintenanceRequest(BaseModel):
    model_config = ConfigDict(extra="forbid")

    kind: Literal["daily", "home", "memory"]
    target_day: str = ""
    metadata: dict[str, JsonValue] = Field(default_factory=dict)
    command_id: str = Field(default="", max_length=128)

File: visualization/src/types/maintenance.ts
/** Maintenance availability and request contracts. */

import type { JsonValue } from "./common";

interface MaintenanceRequestBase {
  metadata?: Record<string, JsonValue>;
  command_id?: string;
}

export type MaintenanceRequest =
  | (MaintenanceRequestBase & {
      kind: "daily" | "home";
      target_day?: never;
    })
  | (MaintenanceRequestBase & {
      kind: "memory";
      target_day: string;
    });

export interface MaintenanceStatus {
  availability: {
    checked_day: string;
    home_pending: boolean;
    home_change_count: number;
    home_skill_memory_count: number;
    memory_pending: boolean;
    memory_days: string[];
  };
}
